internal/pkg/fieldpropagator/analyzer.go

```diff
@@ -44,40 +45,58 @@ var Analyzer = &analysis.Analyzer{
 A field propagator is a function that returns a source field.`,
 	Flags:      config.FlagSet,
 	Run:        run,
-	Requires:   []*analysis.Analyzer{buildssa.Analyzer},
+	Requires:   []*analysis.Analyzer{buildssa.Analyzer, fieldtags.Analyzer},
 	ResultType: reflect.TypeOf(new(ResultType)).Elem(),
 	FactTypes:  []analysis.Fact{new(isFieldPropagator)},
 }
 
+type Analysis struct {
+	pass     *analysis.Pass
+	conf     *config.Config
+	ssaInput *buildssa.SSA
+	tf       fieldtags.TaggedFields
+}
```

Comment:
WDYT about this? I find it makes the function signatures much lighter, and the code more readable.

The name might not be ideal though, I am open to suggestions.